CONTRIBUTING.md

```diff
@@ -208,6 +208,39 @@ of the direction" suddenly has the appearance of "I approve of these changes."
 Instead, add a comment that mentions the usernames that you would like a review
 from. Ask explicitly what you would like feedback on.
 
+## Crate Creation
+
+If your PR includes a new crate, it is required to publish its v0.0.1 version
+before the PR can be merged.  Here're the steps:
+
+* Create a sub-directory for your new crate.
+* Under the newly-created directory, create a Cargo.toml file.  Below is an
+  example template:
+
+```
+[package]
+name = "solana-<PACKAGE_NAME>"
+version = "0.0.1"
+description = "<DESCRIPTION>"
+authors = ["Solana Maintainers <maintainers@solana.foundation>"]
+repository = "https://github.com/solana-labs/solana"
+homepage = "https://solana.com/"
+documentation = "https://docs.rs/solana-<PACKAGE_NAME>"
+license = "Apache-2.0"
+edition = "2021"
+```
+
+* Submit the PR for initial review.  At this moment you should see the crate
+  check fails because the newly created crate has not yet known in public.
```

Comment:
```suggestion
* Submit the PR for initial review.  You should see the crate-check CI job
  fails because the newly created crate is not yet published.
```